### PR TITLE
acrn-config/HV: fix build issue while CDP_ENABLED=y

### PR DESCRIPTION
**Pull request by liuwei987**
- Sync hv reference code that generated by acrn-config tool.
- fix build issue while CDP_ENABLED=y

    Tracked-On: #5902
    Signed-off-by: Wei Liu <weix.w.liu@intel.com>
    Acked-by: Victor Sun <victor.sun@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by jsun26intel** [comment]
The patch need @vijaydhanraj take a look

File: hypervisor/include/arch/x86/board.h [hunk 1]
```diff
@@ -28,8 +28,13 @@ struct vmsix_on_msi_info {
 extern struct dmar_info plat_dmar_info;
 
 #ifdef CONFIG_RDT_ENABLED
-extern struct platform_clos_info platform_l2_clos_array[MAX_PLATFORM_CLOS_NUM];
-extern struct platform_clos_info platform_l3_clos_array[MAX_PLATFORM_CLOS_NUM];
+#ifndef CONFIG_CDP_ENABLED
+#define MAX_CACHE_CLOS_NUM_ENTRIES MAX_PLATFORM_CLOS_NUM
```
Comment (vijaydhanraj) [inline hypervisor/include/arch/x86/board.h:32]: Along with MAX_PLATFORM_CLOS_NUM, can we define MAX_CACHE_CLOS_NUM_ENTRIES, MAX_MBA_CLOS_NUM_ENTRIES and in misc_cfg.h?

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: misc/vm_configs/scenarios/hybrid/nuc7i7dnb/misc_cfg.h [hunk 2]
```diff
@@ -25,4 +25,9 @@
 				"hvlog=2M@0xe00000 "	\
 				"memmap=0x200000$0xe00000"
 
+#ifdef CONFIG_RDT_ENABLED
+#define MAX_MBA_CLOS_NUM_ENTRIES	0U
+#endif
+#define MAX_CACHE_CLOS_NUM_ENTRIES	0U
```
Comment (jsun26intel) [inline misc/vm_configs/scenarios/hybrid/nuc7i7dnb/misc_cfg.h:31]: The MACRO MAX_CACHE_CLOS_NUM_ENTRIES should be in #ifdef CONFIG_RDT_ENABLED.
This change need put line of "uint16_t valid_clos_num = MAX_CACHE_CLOS_NUM_ENTRIES;" and "check_vm_clos_config()" in #ifdef also.

File: hypervisor/arch/x86/rdt.c [hunk 3]
```diff
@@ -22,7 +22,7 @@ const uint16_t hv_clos = 0U;
  * clos value (valid_clos_num) that is common between the resources as
  * each resource's clos max value to have consistent allocation.
  */
-uint16_t valid_clos_num = MAX_PLATFORM_CLOS_NUM;
```
Comment (vijaydhanraj) [inline hypervisor/arch/x86/rdt.c:25]: why change MAX_PLATFORM_CLOS_NUM to MAX_CACHE_CLOS_NUM_ENTRIES? These have 2 different meanings.
MAX_PLATFORM_CLOS_NUM is the common MAX CLOS supported among all RDT resources. But MAX_CACHE_CLOS_NUM_ENTRIES is the number of cache mask entries.  Here I believe it should be MAX_PLATFORM_CLOS_NUM.

File: misc/acrn-config/board_config/misc_cfg_h.py [hunk 4]
```diff
@@ -219,14 +219,29 @@ def generate_file(config):
...
+        print("#else", file=config)
+        print("#define MAX_CACHE_CLOS_NUM_ENTRIES\t{}U".format(max_cache_clos_entries), file=config)
+        print("#endif", file=config)
+        print("#define MAX_MBA_CLOS_NUM_ENTRIES\t{}U".format(max_mba_clos_entries), file=config)
+        print("#endif", file=config)
+    else:
+        print("#ifdef CONFIG_RDT_ENABLED", file=config)
+        print("#define MAX_MBA_CLOS_NUM_ENTRIES\t{}U".format(max_mba_clos_entries), file=config)
+        print("#endif", file=config)
         print("#define MAX_CACHE_CLOS_NUM_ENTRIES\t{}U".format(max_cache_clos_entries), file=config)
```
Comment (vijaydhanraj) [inline misc/acrn-config/board_config/misc_cfg_h.py:240]: As MAX_MBA_CLOS_NUM_ENTRIES is part of CONFIG_RDT_ENABLED define so should MAX_CACHE_CLOS_NUM_ENTRIES be also part of CONFIG_RDT_ENABLED.

File: hypervisor/arch/x86/rdt.c (hunk 3 above)
Comment (dongyaozu) [reply hypervisor/arch/x86/rdt.c:25]: Ｉ have same question: Why we need this change? I didn't see the reasons.

Comment (liuwei987) [reply hypervisor/arch/x86/rdt.c:25]: You are right, i have miss-understanding the 'MAX_PLATFORM_CLOS_NUM' before.
After sync with Vijay and Victor,  they agree with use MAX_COMMON_CLOS macro to instead MAX_PLATFORM_CLOS_NUM macro.

File: misc/acrn-config/board_config/misc_cfg_h.py (hunk 4 above)
Comment (liuwei987) [reply misc/acrn-config/board_config/misc_cfg_h.py:240]: Agree. Will do.

File: misc/vm_configs/scenarios/hybrid/nuc7i7dnb/misc_cfg.h (hunk 2 above)
Comment (liuwei987) [reply misc/vm_configs/scenarios/hybrid/nuc7i7dnb/misc_cfg.h:31]: Agree. Will do.

**Comment by HuiHuangShi** [comment]
4 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:sharing_config filename:/hypervisor/arch/x86/rdt.c func:is_platform_rdt_capable offset:228<br> reason:Procedure is not called or referenced in code analysed. : is_platform_rdt_capable<br><2> config:part_config filename:/hypervisor/arch/x86/rdt.c func:is_platform_rdt_capable offset:228<br> reason:Procedure is not called or referenced in code analysed. : is_platform_rdt_capable<br><3> config:sharing_config filename:/hypervisor/include/arch/x86/rdt.h func: offset:24<br> reason:No real declaration for external variable. : valid_clos_num<br><4> config:part_config filename:/hypervisor/include/arch/x86/rdt.h func: offset:24<br> reason:No real declaration for external variable. : valid_clos_num<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/rdt.c (hunk 3 above)
Comment (dongyaozu) [reply hypervisor/arch/x86/rdt.c:25]: If this is a simple rename, make it a separate patch and PR later with right explaination. Rename is less critical.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/rdt.c (hunk 3 above)
Comment (jsun26intel) [reply hypervisor/arch/x86/rdt.c:25]: @dongyaozu 
This is not only a rename. The previous understanding on MAX_PLATFORM_CLOS_NUM is not correct. We thought the value is decided by native board and it's scenario-irrelevant so put the MACRO in board_info.h.
But actually the value is presenting a common MAX CLOS supported among all RDT resources and this value depends on the setting of CONFIG_CDP_ENABLED, i.e. The MACRO should be scenario-relevant. So we rename the MACRO from MAX_PLATFORM_CLOS_NUM to MAX_COMMON_CLOS and put the definition into scenario specific misc_cfg.h.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**